src/pairwise.jl

```diff
@@ -0,0 +1,166 @@
+function _pairwise!(::Val{:none}, res::AbstractMatrix, f, x, y, symmetric::Bool)
+    m, n = size(res)
+    for j in 1:n, i in 1:m
+        symmetric && i > j && continue
+
+        # For performance, diagonal is special-cased
+        if f === cor && i == j && x[i] === y[j]
+            # If the type isn't concrete, 1 may not be converted to the right type
+            # and the final matrix will have an abstract eltype
+            # (missings are propagated via the second branch, but NaNs are ignored)
```

Comment:
are missings propagated via the second branch? It seems that this branch will be triggered even for missings

Reply:
See https://github.com/JuliaStats/StatsBase.jl/pull/627/files/9c8b3a6599c88537a63f8e773fd111f536de699e#r550921778. I think this needs to be changed.